### ISSUE TITLE
Allow `commentPattern: null` in no-fallthrough

### ISSUE THREAD

**Issue by mdjermanovic** [eslint/eslint#13746]
**What rule do you want to change?**

no-fallthrough

**Does this change cause the rule to produce more or fewer warnings?**

more

**How will the change be implemented? (New option, new default behavior, etc.)?**

The existing option `"commentPattern"` will allow `null` value.

This will be the official way to disable the `commentPattern` feature of this rule for users who prefer explicit `eslint-disable` comments or maybe want to completely disallow falling through case clauses.

A current workaround is to set a pattern that can never match.

Refs: #13260, #13627

**Please provide some example code that this change will affect:**

```js
/* eslint no-fallthrough: "error" */

switch (foo) {
  case 1:
    bar();

  // eslint-disable-next-line no-fallthrough  
  case 2:
    baz();  
}
```

**What does the rule currently do for this code?**

When run with `--report-unused-disable-directives`, ESLint reports:

```
  7:3  error  Unused eslint-disable directive (no problems were reported from 'no-fallthrough')
```

This is because the comment happens to be both an eslint-disable directive and a directive for the `no-fallthrough` rule (since the comment matches the default `commentPattern`) at the same time, so `no-fallthrough` actually doesn't report an error.

**What will the rule do after it's changed?**

If `"commentPattern": null` is set, no comments will match.

I think that this option ideally shouldn't have a default value, but that might be a big breaking change.

**Are you willing to submit a pull request to implement this change?**

Yes.

**Comment by anikethsaha** [comment]
currently  `commentPattern: null` will load the default pattern right!

so if we change it to no pattern, wouldn't this be a breaking change?

**Comment by mdjermanovic** [comment]
That would be certainly a breaking change, but it looks like `null` isn't currently allowed as in [this demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLWZhbGx0aHJvdWdoOiBbXCJlcnJvclwiLCB7IGNvbW1lbnRQYXR0ZXJuOiBudWxsIH1dICovIiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjoxMiwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6eyJjb25zdHJ1Y3Rvci1zdXBlciI6MiwiZm9yLWRpcmVjdGlvbiI6MiwiZ2V0dGVyLXJldHVybiI6Miwibm8tYXN5bmMtcHJvbWlzZS1leGVjdXRvciI6Miwibm8tY2FzZS1kZWNsYXJhdGlvbnMiOjIsIm5vLWNsYXNzLWFzc2lnbiI6Miwibm8tY29tcGFyZS1uZWctemVybyI6Miwibm8tY29uZC1hc3NpZ24iOjIsIm5vLWNvbnN0LWFzc2lnbiI6Miwibm8tY29uc3RhbnQtY29uZGl0aW9uIjoyLCJuby1jb250cm9sLXJlZ2V4IjoyLCJuby1kZWJ1Z2dlciI6Miwibm8tZGVsZXRlLXZhciI6Miwibm8tZHVwZS1hcmdzIjoyLCJuby1kdXBlLWNsYXNzLW1lbWJlcnMiOjIsIm5vLWR1cGUtZWxzZS1pZiI6Miwibm8tZHVwZS1rZXlzIjoyLCJuby1kdXBsaWNhdGUtY2FzZSI6Miwibm8tZW1wdHkiOjIsIm5vLWVtcHR5LWNoYXJhY3Rlci1jbGFzcyI6Miwibm8tZW1wdHktcGF0dGVybiI6Miwibm8tZXgtYXNzaWduIjoyLCJuby1leHRyYS1ib29sZWFuLWNhc3QiOjIsIm5vLWV4dHJhLXNlbWkiOjIsIm5vLWZhbGx0aHJvdWdoIjoyLCJuby1mdW5jLWFzc2lnbiI6Miwibm8tZ2xvYmFsLWFzc2lnbiI6Miwibm8taW1wb3J0LWFzc2lnbiI6Miwibm8taW5uZXItZGVjbGFyYXRpb25zIjoyLCJuby1pbnZhbGlkLXJlZ2V4cCI6Miwibm8taXJyZWd1bGFyLXdoaXRlc3BhY2UiOjIsIm5vLW1pc2xlYWRpbmctY2hhcmFjdGVyLWNsYXNzIjoyLCJuby1taXhlZC1zcGFjZXMtYW5kLXRhYnMiOjIsIm5vLW5ldy1zeW1ib2wiOjIsIm5vLW9iai1jYWxscyI6Miwibm8tb2N0YWwiOjIsIm5vLXByb3RvdHlwZS1idWlsdGlucyI6Miwibm8tcmVkZWNsYXJlIjoyLCJuby1yZWdleC1zcGFjZXMiOjIsIm5vLXNlbGYtYXNzaWduIjoyLCJuby1zZXR0ZXItcmV0dXJuIjoyLCJuby1zaGFkb3ctcmVzdHJpY3RlZC1uYW1lcyI6Miwibm8tc3BhcnNlLWFycmF5cyI6Miwibm8tdGhpcy1iZWZvcmUtc3VwZXIiOjIsIm5vLXVuZGVmIjoyLCJuby11bmV4cGVjdGVkLW11bHRpbGluZSI6Miwibm8tdW5yZWFjaGFibGUiOjIsIm5vLXVuc2FmZS1maW5hbGx5IjoyLCJuby11bnNhZmUtbmVnYXRpb24iOjIsIm5vLXVudXNlZC1sYWJlbHMiOjIsIm5vLXVudXNlZC12YXJzIjoyLCJuby11c2VsZXNzLWNhdGNoIjoyLCJuby11c2VsZXNzLWVzY2FwZSI6Miwibm8td2l0aCI6MiwicmVxdWlyZS15aWVsZCI6MiwidXNlLWlzbmFuIjoyLCJ2YWxpZC10eXBlb2YiOjJ9LCJlbnYiOnt9fX0=) (if I'm not missing something?).

**Comment by anikethsaha** [comment]
Thats cool then. 

+1 for this proposal

**Comment by nzakas** [comment]
I’m not sure this is important enough for a breaking change. It seems like an easier approach would just be to add a note in the rule documentation.

**Comment by nzakas** [comment]
Oops misinterpreted the earlier comment. Even without a breaking change, I’d still prefer not changing the current behavior and just documenting the problem. It seems like the issue is more one of people not understanding how to use the rule rather than a problem with the rule itself.

**Comment by mdjermanovic** [comment]
I realized that the purpose of this rule is actually to mark intentional fall-throughs, not to disallow fall-throughs, so this proposal doesn't make sense.
